src/socket.c

```diff
@@ -4,6 +4,20 @@
 ** See Copyright Notice in mruby.h
 */
 
+#include "mruby.h"
```

Comment:
Why include `mruby.h` here?  What you really need are compile options in `mrbconf.h`?